docs/lang/articles/basic/differences_between_taichi_and_python_programs.md

```diff
@@ -0,0 +1,163 @@
+---
+sidebar_position: 2
+---
+
+# Differences between Taichi and Python programs
+
+Although Taichi uses Python as the frontend, there are some differences between Taichi and Python programs.
+Main differences are:
+
+1. [Taichi only supports return statement outside non-static `if`/`for`/`while` scope in the program](#return-statement)
+2. [Variables defined inside an `if`/`for`/`while` block cannot be accessed outside the block.](#variable-scoping)
+3. [Taichi does not fully support some language features of Python.](#unsupportedpartially-supported-python-language-features)
+  - [Set, list, dictionary and operator `in`](#set-list-dictionary-and-operator-in)
+  - [Comprehensions](#comprehensions)
+  - [Operator `is`](#operator-is)
+
+## Return statement and return type annotation
+
+- If a taichi kernel/function does not have a return statement, it must not have return type annotation.
+- If a taichi kernel has a return statement, it must have return type annotation.
```

Comment:
These two lines are somewhat repeating? May be better with:

In a Taichi kernel, return statement must appear with the existence of return type annotation, vice versa.

Reply:
Make sense